codes/codes.go

```diff
@@ -70,6 +71,7 @@ func (c *Code) UnmarshalJSON(b []byte) error {
 		return fmt.Errorf("nil receiver passed to UnmarshalJSON")
 	}
 
+	b = bytes.Replace(b, []byte("\""), []byte(""), -1)
```

Comment:
nit: this method could be simplified and be more correct, I think, something like:
```
func (c *Code) UnmarshalJSON(b []byte) error {
  var x interface{}
  if err := json.Unmarshal(b, &x); err != nil {
    return err
  }
  if x == nil {
    return fmt.Error("null code value")
  }
  switch x.(type) {
  case string: 
    // ...
  case float64:
    // ...
  }
}
```

Otherwise, this will accept

Reply:
Looks like the values without quotes are acceptable according to this test: 
https://github.com/open-telemetry/opentelemetry-go/blob/master/codes/codes_test.go#L58

Reply:
Unquoted code names are not JSON compatible, right?

Reply:
That's right but I just thought if there's a test for unquoted values maybe these changes could break something.
So if it does not I'll follow your suggestion.

Reply:
I'd like to know how we got here before we un-do it. Does the protocol specification allow this?

Reply:
As I see there's nothing explicit about that: 
https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#set-status

Reply:
This looks like a mistake I made. As commented, I based this off of https://github.com/grpc/grpc-go/blob/bb64fee312b46ebee26be43364a7a966033521b1/codes/codes.go#L218-L244, but the copy of the string-to-code lookup table does not contain quoted codes. I propagated this mistake into that test.

This is indeed a mistake. Unquoted codes are not valid JSON and is incorrect.

I like @jmacd's s suggested refinement. We should definitely update the tests to not check for invalid behavior and instead of testing the `UnmarshalJSON` method directly we could test with `json.Unmarshal`.